src/App.js

```diff
@@ -14,11 +17,137 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    listIsVisible: true,
+    buttonIsVisible: false,
+    products: goodsFromServer,
+    initialState: goodsFromServer,
+    selectedElementLength: 1,
+  }
+
+  showList = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      listIsVisible: !prevState.listIsVisible,
+      buttonIsVisible: !prevState.buttonIsVisible,
```

Comment:
no need to have two properties for this, if the list is visible, that means the button is hidden, so one property should be enough